### PR TITLE
Add support for using a second signer (HW)

### PR DESCRIPTION
**Pull request by fedealconada**
# Description

This PR adds support for using a second signer on the KintoWallet.

This second signer is a Hardware Wallet that could either be "ledger" or "trezor".

I've provided a sample migration (`script/migrations/65-test-wallet-change.s.sol`) with how this can be used. There's no need to execute the migration (but if done, it won't do any change).

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Deployment
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [ ] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [ ] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: src/wallet/KintoWallet.sol [hunk 1]
```diff
@@ -125,9 +127,9 @@ contract KintoWallet is Initializable, BaseAccount, TokenCallbackHandler, IKinto
      * @dev Change signer policy
      * @param policy new policy
      */
-    function setSignerPolicy(uint8 policy) public override onlySelf {
+    function setSignerPolicy(uint8 policy) public override {
```
Comment (rrecuero) [inline src/wallet/KintoWallet.sol:130]: why remove onlySelf?

File: src/wallet/KintoWallet.sol [hunk 2]
```diff
@@ -324,12 +328,9 @@ contract KintoWallet is Initializable, BaseAccount, TokenCallbackHandler, IKinto
             (
                 signerPolicy == SINGLE_SIGNER && owners.length == 1
                     && _verifySingleSignature(owners[0], hashData, userOp.signature) == SIG_VALIDATION_SUCCESS
-            )
-                || (
-                    signerPolicy != SINGLE_SIGNER
-                        && _verifyMultipleSignatures(hashData, userOp.signature) == SIG_VALIDATION_SUCCESS
-                )
+            ) || (signerPolicy != SINGLE_SIGNER && _verifyMultipleSignatures(hashData, userOp.signature))
```
Comment (rrecuero) [inline src/wallet/KintoWallet.sol:331]: I think you still need the == SIG no? need to check if is euqls to 0 otherwise is the opopsite

Comment (ylv-io) [reply src/wallet/KintoWallet.sol:331]: No, because I return bool in `_verifyMultipleSignatures`

Comment (rrecuero) [reply src/wallet/KintoWallet.sol:331]: sounds good, or correct is false got it

File: src/wallet/KintoWallet.sol (hunk 1 above)
Comment (ylv-io) [reply src/wallet/KintoWallet.sol]: This is for testing only. Reverted.

**Pull request merged**